### DISCUSSION TITLE
what is the payments-private-key ?

### DISCUSSION THREAD

**Discussion by wwwehr** [NillionNetwork/discussions#19] [Q&A] [answered]
So part of the run-local-cluster start is that it spins up a local blockchain using foundry/anvil. The values of this (rpc endpoint, wallet, etc) are needed by the nillion client so that it can perform blockchain coordinated tasks.
your config would look something like this:

```
{
    "rpc_endpoint": "http://localhost:45391/",
    "smart_contract_addresses": {
      "blinding_factors_manager": "a513e6e4b8f2a923d98304ec87f64353c4d5c853",
      "payments": "5fc8d32690cc91d4c39d9d3abcbd16989f875707"
    },
    "signer": {
      "wallet": {
        "chain_id": 31337,
        "private_key": "92db14e403b83dfe3df233f83dfa3a0d7096f21ca9b0d6d6b8d88b2b4ec1564e"
      }
    }
  }
  ```
  ^^ but would be dependent on the actual values of your own anvil

**Comment by wwwehr** [comment, marked as answer]
@dnakhoa wrote:

> I believe our current local cluster while running save the config to a yaml file, which doesn’t included all the above information.
> Could you specify the location that I can find the above info.

the command output from `run-local-cluster` should include that extra information.

see how I'm [extracting this config](https://github.com/nillion-oss/builder-resources/blob/main/tools/bootstrap-local-environment.sh#L47) by capturing the output. I suggest you use this bootstrap script or the alternate [dotenv style](https://github.com/nillion-oss/builder-resources/blob/main/py-client/compute-with-permissions/bootstrap-local-environment.sh) that @oceans404 put together.

**Reply by dnakhoa** [reply]
blockchain_rpc_endpoint: http://localhost:44425
chain_id: 31337
payments_sc_address: 5fc8d32690cc91d4c39d9d3abcbd16989f875707
blinding_factors_manager_sc_address: a513e6e4b8f2a923d98304ec87f64353c4d5c853

This is the info our local cluster is storing about the payment config. So another bootstrap script is needed?

**Reply by dnakhoa** [reply]
I believe we were given an outdated guide on how to set up the local cluster, as our current local cluster currently not emitting any information as you said. So whenever we run the cluster the output is like this

![Screen Shot 2024-02-09 at 17 22 41](https://github.com/nillion-oss/builder-resources/assets/40907940/c9f99e35-cc21-4916-a73f-553954a30181)

The logged config doesn't have enough information like you guys have listed here. Is using the new bootstrap file enough to fix the issue, or we need to install a new cluster version.

**Reply by wwwehr** [reply]
@dnakhoa it looks like you should have the info you need. dig it:

![image](https://github.com/nillion-oss/builder-resources/assets/33910651/48e77b3c-094d-4eef-87f1-d189c857f77e)

the values for your payment config are in those two files. just check out the python or javascript examples to see the exact structure for the config itself.
